git_machete/git_operations.py

```diff
@@ -719,3 +723,13 @@ def get_latest_checkout_timestamps(self) -> Dict[str, int]:  # TODO (#110): defa
                 if to_branch not in result:
                     result[to_branch] = int(match.group(1))
         return result
+
+    def check_that_forkpoint_is_ancestor_or_equal_to_head_of_branch(
+            self, forkpoint_sha: str, branch: str) -> None:
+        if not self.is_ancestor_or_equal(
+                earlier_revision=forkpoint_sha,
+                earlier_prefix='',
+                later_revision=branch):
+            raise MacheteException(
+                f"Forkpoint {forkpoint_sha} is not ancestor of or HEAD "
```

Comment:
```suggestion
                f"Fork point {forkpoint_sha} is not ancestor or tip "
```

`HEAD` written in upper case never refers to a tip of a branch. Sometimes `head` in lower case is used in that context by git docs... but this is also confusing. `Tip` is more unambiguous... although given that branches are pointers anyway... `tip of branch X` is kinda equivalent to `branch X` anyway